packages/loader/src/requirejs.ts

```diff
@@ -109,41 +92,122 @@ export function cdn(url: string, min: boolean = true, additionalPaths: { [key: s
+                }
+            }
+            resolve(requirejs.config({
+                context: url,
+                paths
+            }));
+        }
+
+        const oReq = new XMLHttpRequest();
+        oReq.addEventListener("load", reqListener);
+        oReq.open("GET", url + "/loader/package.json");
+        oReq.send();
+    });
+}
+
+function pkgVersion(pkg, id: string): string {
+    const version = pkg.dependencies[id];
+    if (version[0] === "^" || version[0] === "~") {
+        return version.substring(1);
+    }
+    return version;
+}
+
+function pkgUrl(pkg, id: string): string {
+    return `https://unpkg.com/${id}@${pkgVersion(pkg, id)}`;
+}
+
+function unpkgPath(url: string, min: boolean = true, additionalPaths: { [key: string]: string } = {}): Promise<any> {
+    const minStr = min ? ".min" : "";
+    return new Promise((resolve, reject) => {
+        function reqListener() {
+            const pkg = JSON.parse(this.responseText);
+            console.log(`Configuring require from ${pkg.name}@${pkg.version} from ${url}`);
+            const topoJsonUrl = `${pkgUrl(pkg, "@hpcc-js/map")}/TopoJSON`;
+            window.__hpcc_topoJsonFolder = topoJsonUrl;
+            const paths: { [key: string]: string } = {
+                "@hpcc-js/map/TopoJSON": topoJsonUrl,
+                ...additionalPaths
+            };
+            for (const key in pkg.dependencies) {
+                paths[key] = `${pkgUrl(pkg, key)}/dist/index${minStr}`;
+            }
+            resolve(requirejs.config({
+                context: url,
+                paths
+            }));
+        }
+
+        const oReq = new XMLHttpRequest();
+        oReq.addEventListener("load", reqListener);
+        oReq.open("GET", url);
+        oReq.send();
+    });
+}
+
+export function unpkgVersion(version: string = "", min: boolean = true, additionalPaths: { [key: string]: string } = {}): Promise<any> {
+    if (version) version = "@" + version;
+    return unpkgPath(`https://unpkg.com/@hpcc-js/loader${version}/package.json`, min, additionalPaths);
+}
+
 export function unpkg(min: boolean = true, additionalPaths: { [key: string]: string } = {}): any {
+    console.log("Deprecated - please use 'amd'");
     window.__hpcc_topoJsonFolder = "https://unpkg.com/@hpcc-js/map/TopoJSON";
     return cdn("https://unpkg.com/@hpcc-js", min, additionalPaths);
 }
 
-function local(additionalPaths: { [key: string]: string }, min: boolean = false): any {
-    const config = parseScriptUrl(true);
-    window.__hpcc_topoJsonFolder = `${config.libUrl}/map/TopoJSON`;
+export function dev(additionalPaths: { [key: string]: string } = {}): any {
+    window.__hpcc_topoJsonFolder = `${hostUrl}/map/TopoJSON`;
     const thirdPartyPaths: { [key: string]: string } = {};
     for (const key in npmPackages) {
-        thirdPartyPaths[key] = `${config.node_modulesUrl}/${npmPackages[key]}`;
+        thirdPartyPaths[key] = `${hostUrl}/../node_modules/${npmPackages[key]}`;
```

Comment:
Can this path assumption be made here? Seems like this belongs at the application layer.

Reply:
Its only within the context of the "dev" mode, so should be fine...